ipset.go

```diff
@@ -332,6 +333,77 @@ type IPSet struct {
 	rr []IPRange
 }
 
+// ParseIPSet parses a comma-separated list of IP ranges into an IPSet.
+// If an invalid IPRange is provided, an empty IPSet and an error are returned.
+// The provided list need not be representative of a true set; ranges may
+// overlap and duplicates are allowed.
+// An empty string produces an empty set.
+func ParseIPSet(s string) (*IPSet, error) {
+	var (
+		b   IPSetBuilder
+		r   IPRange
+		err error
+	)
+	if len(s) == 0 {
+		return b.IPSet()
+	}
+	rss := strings.Split(s, ",")
+	for _, rs := range rss {
+		r, err = ParseIPRange(rs)
+		if err != nil {
+			return &IPSet{}, fmt.Errorf("invalid IP range %q in set %q", rs, s)
+		}
+		b.AddRange(r)
+	}
+	return b.IPSet()
```

Comment:
A `nil, err` return definitely feels most idiomatic, especially in the case of an all-or-nothing action. I'm struggling to come up with a use case where the resulting empty set would be put to valid use in case of an error.

Reply:
Went ahead and changed it to a `nil, err` return, but am happy to change it back if desired.